src/module/dy.js

```diff
@@ -31,22 +31,40 @@ module.exports = ({ call, getService, logger }) => ({
         ...item
       };
     };
+    const extractKey = (item) => model.schema.KeySchema
+      .map(({ AttributeName: attr }) => attr)
+      .reduce((prev, cur) => Object.assign(prev, { [cur]: item[cur] }), {});
```

Comment:
```suggestion
      .reduce((prev, cur) => Object.assign(prev, { [cur]: item[cur] }), {});
```
use return here and assign to object. Assign is just more expensive